### PR TITLE
use clap 3.0.0-rc.5 to unify the cli arg parsing

### LINKED ISSUE #3504: Clap  `--single` is not compatible
Ubuntu platform uses `databend-query-standalone-embedded-meta.sh` which doesn't depend on `--single` option. So the root cause is the change for `--single` is not compatible. It used to support `--single=true` but now it doesn't.

To me, it's OK to use `--single` only, if so, we need to update scripts under `scripts/deploy`.

ping @BohuTANG for review.

_Originally posted by @Xuanwo in https://github.com/datafuselabs/databend/issues/3499#issuecomment-995601515_

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

use clap 3 to unify the cli arguments parsing

## Changelog

- Improvement

## Related Issues

Close #3504 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -19,18 +19,18 @@ path = "src/bin/bendctl.rs"
 
 [dependencies]
 # Workspace dependencies
-databend-query = {path = "../query"}
-common-base = {path = "../common/base" }
+databend-query = { path = "../query" }
```
Comment (BohuTANG) [inline cli/Cargo.toml:22]: Cool.
These pretty format is by tool or hand?

File: common/meta/raft-store/src/config.rs [hunk 2]
```diff
@@ -29,109 +29,68 @@ pub const KVSRV_BOOT: &str = "KVSRV_BOOT";
...
-    )]
+    /// The listening host for metadata communication.
+    #[clap(long, env = KVSRV_API_HOST, default_value = "127.0.0.1")]
     pub raft_api_host: String,
 
-    #[structopt(
-        long,
-        env = KVSRV_API_PORT,
-        default_value = "28004",
-        help = "The listening port for metadata communication"
```
Comment (BohuTANG) [inline common/meta/raft-store/src/config.rs:50]: Why remove help here?

Comment (BohuTANG) [reply common/meta/raft-store/src/config.rs:50]: The `help` will be helpful with:
databend-meta -h

File: cli/Cargo.toml (hunk 1 above)
Comment (koushiro) [reply cli/Cargo.toml:22]: by vscode plugin

File: common/meta/raft-store/src/config.rs (hunk 2 above)
Comment (koushiro) [reply common/meta/raft-store/src/config.rs:50]: the doc will be help info

File: cli/Cargo.toml (hunk 1 above)
Comment (BohuTANG) [reply cli/Cargo.toml:22]: Emm, what's the plugin name? Thanks :D

Comment (koushiro) [reply cli/Cargo.toml:22]: Even Better TOML

**Comment by BohuTANG** [comment]
This patch seems not work on macOS:
```
Waiting on databend-meta 10 seconds...
error: The value '=true' was provided to '--single' but it wasn't expecting any more values

databend-meta [OPTIONS]

For more information try --help
```

CI failure:
https://github.com/datafuselabs/databend/runs/4545281445?check_suite_focus=true

**Comment by Xuanwo** [comment]
Ubuntu platform uses `databend-query-standalone-embedded-meta.sh` which doesn't depend on `--single` option. So the root cause is the change for `--single` is not compatible. It used to support `--single=true` but now it doesn't.

To me, it's OK to use `--single` only, if so, we need to update scripts under `scripts/deploy`.

ping @BohuTANG for review.

**Comment by koushiro** [comment]
I could add `parse(try_from_str)` attribute for the `single` field to keep compatible

**Comment by BohuTANG** [comment]
> Ubuntu platform uses `databend-query-standalone-embedded-meta.sh` which doesn't depend on `--single` option. So the root cause is the change for `--single` is not compatible. It used to support `--single=true` but now it doesn't.
> 
> To me, it's OK to use `--single` only, if so, we need to update scripts under `scripts/deploy`.
> 
> ping @BohuTANG for review.

@koushiro Could you have plan to fix it in this PR? Thank you.

**Comment by BohuTANG** [comment]
`--single` is better, we can fix it in the script.

**Comment by koushiro** [comment]
> `--single` is better, we can fix it in the script.

I seem to have seen the same issues before, I will fix them together later.
I'm outside for dinner now :p

**Comment by BohuTANG** [comment]
Still have some issues with databend-meta:
```
The databend-query is not killed. force killing.
databend-query: no process found
=== boot new metasrv cluster of 3
Start Meta service HA cluster(3 nodes)...
error: Found argument 'true' which wasn't expected, or isn't valid in this context

USAGE:
    databend-meta [OPTIONS]
```

https://github.com/datafuselabs/databend/runs/4555094771?check_suite_focus=true

**Review by PsiACE — APPROVED** [review]
LGTM

**Pull request merged**